core/src/test/java/io/onetable/ITOneTableClient.java

```diff
@@ -145,6 +151,8 @@ private SourceClientProvider<?> getSourceClientProvider(TableFormat sourceTableF
       return hudiSourceClientProvider;
     } else if (sourceTableFormat == TableFormat.DELTA) {
       return deltaSourceClientProvider;
+    } else if (sourceTableFormat == TableFormat.ICEBERG) {
```

Comment:
Can we change this method to instead create and return the SourceClientProvider instead of creating all 3 instances in the `BeforeEach`?

Reply:
Good catch. Instead of creating 3 instances we now create 1 as required.